faunadb-java/src/main/java/com/faunadb/client/FaunaClient.java

```diff
@@ -228,6 +229,26 @@ public FaunaClient newSessionClient(String secret) {
     return performRequest(json.valueToTree(expr), timeout);
   }
 
+  /**
+   * Issues a Query to FaunaDB with extra information
+   * <p>
+   * Queries are constructed by the helper methods in the {@link com.faunadb.client.query.Language} class.
+   * <p>
+   * Responses are represented as structured tree where each node is a {@link Value} instance.
+   * {@link Value} instances can be converted to native types. See {@link Value} class for details.
+   *
+   * @param expr the query to be executed.
+   * @param timeout the timeout for the current query. It replaces the timeout value set for this
+   *                {@link FaunaClient} (if any), for the scope of this query. The timeout value
+   *                has milliseconds precision.
+   * @return a {@link CompletableFuture} containing the root node of the response tree.
+   * @see MetricsResponse
+   * @see com.faunadb.client.query.Language
+   */
+  public CompletableFuture<MetricsResponse> queryWithMetrics(Expr expr, Optional<Duration> timeout) {
```

Comment:
I think it needs to be exposed in the Scala client as well. WDYT?

Reply:
yeah, I wanted to add it there as well, will do